x86_att/p256/p256_montjmixadd.S

```diff
@@ -470,14 +472,14 @@ S2N_BN_SYMBOL(p256_montjmixadd):
 // Save registers and make room on stack for temporary variables
 // Put the input y in %rbp where it lasts as long as it's needed.
 
-        pushq  %rbx
-        pushq  %rbp
-        pushq  %r12
-        pushq  %r13
-        pushq  %r14
-        pushq  %r15
+        CFI_PUSH(%rbx)
+        CFI_PUSH(%rbp)
+        CFI_PUSH(%r12)
+        CFI_PUSH(%r13)
+        CFI_PUSH(%r14)
+        CFI_PUSH(%r15)
 
-        subq    $NSPACE, %rsp
+        CFI_DEC_RSP(NSPACE)
 
         movq    %rdx, %rbp
```

Comment:
Here is one of the few places outside of SM2-specific code where RBP is changed, AFAICT.

Reply:
Actually I think there are many more `RBP` changes, since very often it's hidden under macros. If you look at the `objdump` output and `MAYCHANGE` fields in the proof, you can see them all. Try for example:
`grep define x86/*/*.S | grep rbp$`
In particular, it's often used as a zero for `ADCX`/`ADOX` which do not allow immediate arguments.

Reply:
Indeed, I think I overlooked the other uses because it is referenced as `ebp` elsewhere, presumably as an optimization (e.g. `xor ebp, ebp`).

In any case, I seems like I was misremembering, since CFI will use `rsp` as the default base instead of `rbp` like I thought, so AFAICT we don't have to pay any special attention to rbp.

Reply:
Ah yes indeed, there is also the `ebp` optimization. This codebase started with NASM, which makes a few optimizations such as `xor ebp, ebp` for you automatically if you write `rbp` everywhere. But we made it explicit when switching to the GNU assembler (or its clang counterpart).